### DISCUSSION TITLE
import from excel

### DISCUSSION THREAD

**Discussion by juliermefelix** [serenity-is/Serenity#6532] [Help] [answered]
Dear,
I'm using the newest version of serenity with .net 6 and taking advantage of the MVC code to import from Excel, but I'm having some problems.

![image](https://user-images.githubusercontent.com/7516443/189635056-82dbf63f-4f4d-4e9b-9335-441daff46d1a.png)

In MVC I used repository and now I tried to use the controller but it doesn't accept
![image](https://user-images.githubusercontent.com/7516443/189635370-e3d86f49-6e21-43e2-9049-7548c10dc162.png)

Previously the example is in the basic samples, now it is in startsharp, so I can't compare the codes.

**Comment by juliermefelix** [comment, marked as answer]
Sorted out,
I changed it to the code below.
![image](https://user-images.githubusercontent.com/7516443/189948243-fd58384d-60e1-4605-acf6-cac451002393.png)
I changed it to xxxSaveHandler and insert "context" between the ()
![image](https://user-images.githubusercontent.com/7516443/189948552-2a8b3b19-b3b4-4ed2-8ac2-ca88dc4dae62.png)

**Reply by gfo2007** [reply]
@juliermefelix could you be kind to share your excel import code with me? I want to implement import from excel but finding it hard to do so.

Regards

**Reply by VictorTomaili** [reply]
If it's startsharp excel import sample, you cannot share the code because of the license.

**Reply by juliermefelix** [reply]
As mentioned above I used the MVC code, I only changed the line that was causing the problem, pointing to the xxxSaveHandler(context).

**Reply by VictorTomaili** [reply]
It's okay @juliermefelix, I just point the restrictions for @gfo2007 regarding request of the excel import example. There is no problem about your question.

**Reply by gfo2007** [reply]
@juliermefelix is your code from StartSharp or Serenity open source?

**Reply by juliermefelix** [reply]
@gfo2007 
I used the free version of serenity and the MVC import making some changes. Post your doubts here and we'll help you.
